src/sentry/api/endpoints/organization_member_index.py

```diff
@@ -43,3 +74,101 @@ def get(self, request, organization):
             on_results=lambda x: serialize(x, request.user),
             paginator_cls=OffsetPaginator,
         )
+
+    def post(self, request, organization):
+        """
+        Add a Member to Organization
+        ````````````````````````````
+
+        Invite a member to the organization.
+
+        :pparam string organization_slug: the slug of the organization the member will belong to
+        :param string email: the email address to invite
+        :param string role: the role the new member
+        :param array teams: the slugs of the teams the member should belong to.
+
+        :auth: required
+        """
+        # TODO: If the member already exists, should this still update the role and team?
+        # For now, it doesn't, but simply returns the existing object
+
+        serializer = OrganizationMemberSerializer(data=request.DATA)
+
+        if not serializer.is_valid():
+            return Response(serializer.errors, status=400)
+
+        result = serializer.object
+
+        can_admin, allowed_roles = get_allowed_roles(request, organization)
+
+        if not result['role'] in {r.id for r in allowed_roles}:
+            return Response(
+                '{"error": "You do not have permission to invite that role."}',
+                status=403
+            )
+
+        # This is needed because `email` field is case sensitive, but from a user perspective,
+        # Sentry treats email as case-insensitive (Eric@sentry.io equals eric@sentry.io).
+        try:
+            existing = OrganizationMember.objects.filter(
+                organization=organization,
+                user__email__iexact=result['email'],
+                user__is_active=True,
+            )[0]
+
+        except IndexError:
+            pass
+        else:
+            messages.add_message(
+                request, messages.INFO,
+                _('The organization member %s already exists.') % result['email']
+            )
+            return Response(serialize(existing), status=200)
+
+        om = OrganizationMember(
+            organization=organization,
+            email=result['email'],
+            role=result['role'])
+
+        if settings.SENTRY_ENABLE_INVITES:
+            om.token = om.generate_token()
+
+        sid = transaction.savepoint(using='default')
+        try:
+            om.save()
+
+        except IntegrityError:
+            transaction.savepoint_rollback(sid, using='default')
+            messages.add_message(
+                request, messages.INFO,
+                _('The organization member %s already exists.') % result['email']
+            )
+            return Response(serialize(OrganizationMember.objects.get(
+                email__iexact=om.email,
+                organization=organization,
+            )), status=200)
+
+        messages.add_message(
```

Comment:
this case is a little trickier, since i think we need to redirect for it to not be weird.. what if you did something like show an indicator in react and delay the redirect for a second or two? we can also check w ben or chris maybe to see if they think that'd be terrible